### PR TITLE
Examples Browser fixes

### LINKED ISSUE #3958: Two copies of `wasm-loader.js`
There are currently two copies of `wasm-loader.js` in the `examples` folder:

* `examples/lib/wasm-loader.js`
* `examples/src/wasm-loader.js`

Are both needed?

### PR DESCRIPTION
**Pull request by ellthompson**
This PR readies the examples browser for release alongside the 1.52.0 engine.

Changes include:
- The `pc` type definitions are now displaying correctly in the code editor in both javascript and typescript modes.
- Removed a duplicate of the wasm loader file (Fixes #3958).
- All examples now correctly respond the resize events (Fixes ##3959).
- Updated the process for local development of examples and included new documentation for this in the README file.
- Code editing in the browser now correctly works in TypeScript mode.
- Fixed an issue that prevented the mini-stats example from loading.
- Updated all paths to static files to ensure parity when loading an example in both the main application and as an individual iframe.

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: examples/scripts/iframe/index.mustache [hunk 1]
```diff
@@ -60,6 +62,23 @@
                 if (window.top._showMiniStats !== undefined) miniStats.enabled = window.top._showMiniStats;
             });
 
+            var canvasContainerElement = canvas.parentElement;
+            canvas.setAttribute('width', window.innerWidth + 'px');
+            canvas.setAttribute('height', window.innerHeight + 'px');
+            var resizeTimeout = null;
+            pc.app.setCanvasResolution(pc.RESOLUTION_AUTO);
```
Comment (willeastcott) [inline examples/scripts/iframe/index.mustache:69]:
```suggestion
            app.setCanvasResolution(pc.RESOLUTION_AUTO);
```

File: examples/src/app/code-editor.tsx [hunk 2]
```diff
@@ -37,7 +37,11 @@ const CodeEditor = (props: CodeEditorProps) => {
 
     const beforeMount = (monaco: any) => {
         monaco.languages.typescript.typescriptDefaults.addExtraLib(
-            playcanvasTypeDefs,
+            playcanvasTypeDefs + ' export as namespace pc;',
```
Comment (willeastcott) [inline examples/src/app/code-editor.tsx:40]: Doesn't this suggest there's something missing from the `playcanvas.d.ts` file? `export as namespace pc` used to be in the old types but I thought maybe it was unneeded. I can add it back if necessary?

Comment (ellthompson) [reply examples/src/app/code-editor.tsx:40]: Yeah that'd be good as this used to be present.

File: examples/src/app/helpers/formatters.mjs [hunk 3]
```diff
@@ -34,7 +34,11 @@ const getTypeScriptFunctionFromText = (text) => {
 };
 
 const getInnerFunctionText = (text) => {
-    text = text.substring(text.indexOf("{") + 1);
+    if (text.includes(': void {')) {
+        text = text.substring(text.indexOf(": void {") + 8);
```
Comment (mvaligursky) [inline examples/src/app/helpers/formatters.mjs:38]: the `+8` bit here does not fill me with confidence. Perhaps this should be refactored to be more self-explanatory, or a comment should be added?

Comment (ellthompson) [reply examples/src/app/helpers/formatters.mjs:38]: Yeah it's definitely brittle, unfortunately I've found no great way to template these example files so I can extract various parts in a less brittle way. I'll try and update this to be more self explanatory though.

File: examples/src/app/code-editor.tsx (hunk 2 above)
Comment (ellthompson) [reply examples/src/app/code-editor.tsx]: I've updated the rollup config as suggested.

**Review by willeastcott — APPROVED** [review]

**Pull request merged**